From fa387da2f416667b14de57686ab55489d3325677 Mon Sep 17 00:00:00 2001
From: Mario Sanchez Prada <mario@endlessm.com>
Date: Thu, 28 Jan 2016 13:53:08 +0000
Subject: [PATCH 12/22] memory: Enable the tab discards feature

This allows manually discarding tabs from chrome://discards as well
as automatic tab discards once a certain level of "memory pressure"
is reached, using Endless's implementation of MemoryPressureMonitor.

[endlessm/eos-shell#1331]
---
 chrome/browser/resource_coordinator/tab_manager.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/browser/resource_coordinator/tab_manager.cc b/chrome/browser/resource_coordinator/tab_manager.cc
index a85f262655796..b0371972d8532 100644
--- a/chrome/browser/resource_coordinator/tab_manager.cc
+++ b/chrome/browser/resource_coordinator/tab_manager.cc
@@ -166,7 +166,8 @@ void TabManager::Start() {
 
 // MemoryPressureMonitor is not implemented on Linux so far and tabs are never
 // discarded.
-#if defined(OS_WIN) || defined(OS_MAC) || defined(OS_CHROMEOS)
+#if defined(OS_WIN) || defined(OS_MAC) || defined(OS_CHROMEOS) || \
+    defined(OS_LINUX)
   // Don't handle memory pressure events here if this is done by
   // PerformanceManager.
   if (!base::FeatureList::IsEnabled(
-- 
2.26.2

